sparse/finch_backend/__init__.py

```diff
@@ -3,11 +3,93 @@
 except ModuleNotFoundError as e:
     raise ImportError("Finch not installed. Run `pip install sparse[finch]` to enable Finch backend") from e
 
-from finch import Tensor, astype, permute_dims
+from finch import (
+    Dense,
+    DenseStorage,
+    Element,
+    Pattern,
+    RepeatRLE,
+    SparseByteMap,
+    SparseCOO,
+    SparseHash,
+    SparseList,
+    SparseVBL,
+    Storage,
+    Tensor,
```

Comment:
I would keep the API as Finch-agnostic as possible.

Reply:
But to define e.g. `Storage(Dense(...))` we would need to `import finch` separately. I think for now we can keep it in the `finch_backend` namespace, WDYT?

Reply:
@hameerabbasi Done! I removed Finch-specific objects from the import list.